### PR TITLE
Adding NIC statistics collection and reporting

### PR DESCRIPTION
**Pull request by nycaleksey**

### CODE REVIEW

**Comment by mimugmail** [comment]
And this works for you? It didn't display anything in my Virtualbox VM

**Comment by nycaleksey** [comment]
Yeah, works just fine - it started reporting NIC statistics to influxdb and I started seeing it via Grafana.

Here’s evidence inputs.net <http://inputs.net/> works in FreeBSD - https://digi-merc.org/blog/2017-01-14-pfsense-influxdb-metrics-collection/ <https://digi-merc.org/blog/2017-01-14-pfsense-influxdb-metrics-collection/>

**Comment by nycaleksey** [comment]
Can you check telegraf.conf to make sure [[inputs.net]] was added ?

**Comment by mimugmail** [comment]
Sure, it's there, but the graphs are emtpy:
image

I asked around if it works for other with no reply, so I didn't add it.
Do you run hardware appliance?

**Comment by nycaleksey** [comment]
Strange…
No, this is not an appliance - just a PC with two Intel NICs. Here’s how my graph looks:

**Comment by mimugmail** [comment]
Ok, let me install it on a hardware appliance and check again (Thursday)
You can apply your patches on your own so it will survive the reboot until this is fixed:

```
opnsense-patch -c plugins 05ed716
opnsense-patch -c plugins dba97ca
opnsense-patch -c plugins 2245bec
```

**Comment by nycaleksey** [comment]
Thanks a lot!

**Comment by mimugmail** [comment]
@fabianfrz As discussed with @nycaleksey I'll test a hw setup next Thursday, it depends how we'll change the default for network. I prefer 0 but we'll see ...

**Comment by nycaleksey** [comment]
I agree, 0 should be fine.

**Comment by ppx17** [comment]
I've tested this by adding [[inputs.net]] manually to my telegraf.conf and can confirm it works. My OPNsense instance is running on VMware.

**Comment by mimugmail** [comment]
@nycaleksey Please change the default to zero so this can be merged, and don't forget to bump the version number.

**Comment by nycaleksey** [comment]
I just did. Apologies if I missed something - just started learning Git and GitHub

File: net-mgmt/telegraf/src/opnsense/mvc/app/models/OPNsense/Telegraf/Input.xml [hunk 1]
```diff
@@ -51,5 +51,9 @@
             <default>1</default>
             <Required>N</Required>
         </system>
+        <network type="BooleanField">
+            <default>1</default>
```
Comment (mimugmail) [inline net-mgmt/telegraf/src/opnsense/mvc/app/models/OPNsense/Telegraf/Input.xml]: This should be zero ...

File: net-mgmt/telegraf/src/opnsense/service/templates/OPNsense/Telegraf/telegraf.conf [hunk 2]
```diff
@@ -106,4 +106,8 @@
 [[inputs.system]]
 {% endif %}
 
+{% if helpers.exists('OPNsense.telegraf.input.network') and OPNsense.telegraf.input.network == '0' %}
```
Comment (mimugmail) [inline net-mgmt/telegraf/src/opnsense/service/templates/OPNsense/Telegraf/telegraf.conf]: And this revert to 1 please :)

**Comment by nycaleksey** [comment]
I know - this is ridiculous. Sorry, it was a crazy day. Will fix and make proper changes.

**Comment by mimugmail** [comment]
Great, now go to makefile and bump version :)

**Comment by nycaleksey** [comment]
Done

**Comment by nycaleksey** [comment]
Thank you!

**Comment by mimugmail** [comment]
@fabianfrz @fichtner You can merge this now, thanks to all 👍

**Comment by fichtner** [comment]
Thanks guys. Greetings from Thailand :)

**Pull request merged**